src/login/controller.js

```diff
@@ -38,50 +38,56 @@ function LoginController(CardshifterServerAPI, $scope, $location, $rootScope) {
-                                playerIndex: null,
-                                game: {
-                                    id: null,
-                                    mod: null
-                                }
-                            };
+                    CardshifterServerAPI.setMessageListener({
+                        "loginresponse": function(welcome) {
+                            if(welcome.status === SUCCESS && welcome.message === "OK") {
+                                // taking the easy way out
+                                window.currentUser = {
+                                    username: $scope.username,
+                                    id: welcome.userId,
+                                    playerIndex: null,
+                                    game: {
+                                        id: null,
+                                        mod: null
+                                    }
+                                };
 
-                            // for remembering form data
-                            for(var storage in loginStorageMap) {
-                                if(loginStorageMap.hasOwnProperty(storage)) {
-                                    localStorage.setItem(storage, $scope[loginStorageMap[storage]]);
+                                // for remembering form data
+                                for(var storage in loginStorageMap) {
+                                    if(loginStorageMap.hasOwnProperty(storage)) {
+                                        localStorage.setItem(storage, $scope[loginStorageMap[storage]]);
+                                    }
                                 }
-                            }
 
-                            $rootScope.$apply(function() {
                                 $location.path("/lobby");
-                            });
-                        } else {
-                            console.log("server messsage: " + welcome.message);
-                            $scope.loggedIn = false;
-                            $scope.$apply();
+                            } else {
+                                console.log("server messsage: " + welcome.message);
+                                $scope.loggedIn = false;
+                                $scope.$apply();
+                            }
                         }
-                    }, ["loginresponse"]);
+                    }, $scope);
                     CardshifterServerAPI.sendMessage(login);
 
                 } catch(e) {
                     // notify the user that there was an issue logging in (loginmessage issue)
+                    ErrorCreator.create("There was an error logging in");
+
                     console.log("LoginMessage error(error 2): " + e);
                     $scope.loggedIn = false;
                     $scope.$apply();
                 }
             } else {
+                ErrorCreator.create("Please enter a username");
```

Comment:
This code has too many nesting levels. Maybe things can be separated into functions to remedy this. It's currently really hard to see what `if` this is an `else` to.